### PR TITLE
Fix decoding issue (#158)

### PR DESCRIPTION
**Pull request by lquerel**
Attributes with empty keys were already ignored, but not attributes with no value. In this PR, I consider an attribute with no value as malformed or invalid, so they are ignored. 
Let me know if this is not the behavior you expected.

Closes: #158

### CODE REVIEW

File: pkg/otel/common/arrow/attributes.go [hunk 1]
```diff
@@ -236,10 +236,14 @@ func (c *Attributes16Accumulator) AppendWithID(parentID uint16, attrs pcommon.Ma
 	}
 
 	attrs.Range(func(key string, v pcommon.Value) bool {
+		// Attribute without key or value are ignored.
 		if key == "" {
 			// Skip entries with empty keys
 			return true
 		}
+		if v.Type() == pcommon.ValueTypeEmpty {
```
Comment (moh-osman3) [inline pkg/otel/common/arrow/attributes.go]: Should other append functions in this file that skip empty keys follow this convention?

Comment (moh-osman3) [reply pkg/otel/common/arrow/attributes.go]: Also is there a meaningful difference between `v.Type() == pcommon.ValueTypeEmpty` and `v.Type() == pcommon.ValueTypeStr && v.Str() == ""`

**Review by moh-osman3 — APPROVED** [review]
Left a couple non-blocking questions

File: pkg/otel/common/arrow/attributes.go (hunk 1 above)
Comment (lquerel) [reply pkg/otel/common/arrow/attributes.go]: First question: Yes good catch

Second question: Empty strings are actually detected and encoded in Arrow. So empty string won't trigger the same bug.

**Review by jmacd — APPROVED** [review]

**Pull request merged**